### PR TITLE
Blocks: Add `of` prop to `Subtitle`

### PR DESCRIPTION
**Pull request by joaonunomota**
Closes part of #22490

## What I did

- Add support for `of` prop via `useOf` with "meta" as only valid type
- Default to extracting content from parameters.docs.subtitle
- Deprecate extracting content from parameters.componentSubtitle
- Add stories for several possible applications of Subtitle
- Update docs and migration files
- Fix typo in EmptyExample

## How to test

1. Run `yarn storybook:blocks`
2. Open Storybook in your browser
3. Access Subtitle stories

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [x] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: docs/api/doc-block-subtitle.md [hunk 1]
```diff
@@ -24,10 +24,16 @@ import { Subtitle } from '@storybook/blocks';
 
 `Subtitle` is configured with the following props:
 
+### `of`
```
Comment (kylegach) [inline docs/api/doc-block-subtitle.md:27]: Thanks for updating the API reference! ❤️

Could you please keep these props in alphabetical order?

Comment (joaonunomota) [reply docs/api/doc-block-subtitle.md]: Sorted :)

**Comment by joaonunomota** [comment]
@JReinhold Hi! Is there anything that I can update / fix for a pull request review? It would be nice if this PR and #23728 could go out with #23849.

**Comment by JReinhold** [comment]
Sorry @joaonunomota I've missed this. I'll review this early next week

**Comment by ndelangen** [comment]
@JReinhold please review?

**Review by JReinhold — APPROVED** [review]
Thank you for the fantastic work, I'm sorry it took so long for me to properly review it. <3

**Pull request merged**